PolyEngine/Engine/Src/TransformComponent.cpp

```diff
@@ -0,0 +1,32 @@
+#include "TransformComponent.hpp"
+
+using namespace Poly;
+
+//------------------------------------------------------------------------------
+void TransformComponent::SetLocalTranslation(Vector position) {
+	LocalTranslation = position;
+	Dirty = true;
+}
+
+//------------------------------------------------------------------------------
+void TransformComponent::SetLocalRotation(EulerAngles angles) { 
+	LocalRotation = Quaternion(angles); 
+}
+
+//------------------------------------------------------------------------------
+void TransformComponent::SetLocalScale(Vector scale) {
+	LocalScale = scale;
+	Dirty = true;
+}
+
+//------------------------------------------------------------------------------
+Matrix TransformComponent::GetLocalTransformationMatrix() {
```

Comment:
this method should return `const Matrix&` and be const itself:
`const Matrix& TransformComponent::GetLocalTransformationMatrix() const`